gh-deployments-environments/helper.sh

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env bash
+
+set -e
+
+CMD=${1:-};
+
+
+print_syntax() {
+  echo "helper.sh <command> [args...]"
+  echo ""
+  echo "Required environment variables:"
+  echo "  GITHUB_TOKEN  GitHub token used when calling API"
+  echo ""
+  echo "Available commands:"
+  echo "  help"
+  echo "    Prints this screen"
+  echo ""
+  echo "  create-deployment <owner> <repo> <env> <ref> <output-file-for-id>"
+  echo "    Creates new deployment"
+  echo ""
+  echo "  create-deployment-status <owner> <repo> <deployment-id> <status>"
+  echo "    Creates deployment status"
+  echo ""
+  echo "  delete-environment <owner> <repo> <env>"
+  echo "    Deletes environment and all deployments and statuses related to it"
+  echo ""
+}
+
+
+check_env_vars() {
+	if [[ -z "${GITHUB_TOKEN}" ]]; then
+		echo "!!! GITHUB_TOKEN environment variable is not set";
+		print_syntax;
+		exit 1;
+	fi
+}
+
+check_owner_and_repo() {
+  if [[ ! "${1}" =~ ^[a-zA-Z0-9_\-]{1,50}$ ]]; then
+    echo "!!! owner is invalid"
+    exit 1
+  fi
+  if [[ ! "${2}" =~ ^[0-9a-zA-Z_\-]{1,50}$ ]]; then
+    echo "!!! repo is invalid"
+    exit 1
+  fi
+}
+
+check_env() {
+  if [[ ! "${1}" =~ ^[0-9a-zA-Z_\-]{1,100}$ ]]; then
+    echo "!!! env is invalid"
+    exit 1
+  fi
+  if [[ "${1}" == "mainnet" || "${1}" == "testnet" || "${1}" == "devnet" ]]; then
+    echo "!!! env cannot be one of 'mainnet', 'testnet' or 'devnet'"
+    exit 1
+  fi
+}
+
+check_ref() {
+  if [[ ! "${1}" =~ ^[0-9a-zA-Z_.:\-]{1,100}$ ]]; then
+    echo "!!! ref is invalid"
+    exit 1
+  fi
+}
+
+check_deployment_id() {
+  if [[ ! "${1}" =~ ^[0-9]{1,32}$ ]]; then
+    echo "!!! deployment-id is invalid"
+    exit 1
+  fi
+}
+
+check_status() {
+  if [[ ! "${1}" =~ ^[a-z_\-]{1,32}$ ]]; then
+    echo "!!! status is invalid"
+    exit 1
+  fi
+}
+
+make_github_api_call() {
+  if [[ "${1}" == "POST" ]]; then
+    curl -L \
+      -X "${1}" \
+      -H "Accept: application/vnd.github+json" \
+      -H "Authorization: Bearer ${GITHUB_TOKEN}"\
+      -H "X-GitHub-Api-Version: 2022-11-28" \
+      https://api.github.com/repos/${2}/${3}/${4} \
+      -d "${5}" > "${6}"
+  fi
+  if [[ "${1}" == "GET" ]]; then
+    curl -L \
+      -H "Accept: application/vnd.github+json" \
+      -H "Authorization: Bearer ${GITHUB_TOKEN}"\
+      -H "X-GitHub-Api-Version: 2022-11-28" \
+      https://api.github.com/repos/${2}/${3}/${4} > "${6}"
+  fi
+  if [[ "${1}" == "DELETE" ]]; then
+    curl -L \
+      -X "${1}" \
+      -H "Accept: application/vnd.github+json" \
+      -H "Authorization: Bearer ${GITHUB_TOKEN}"\
+      -H "X-GitHub-Api-Version: 2022-11-28" \
+      https://api.github.com/repos/${2}/${3}/${4} > "${6}"
```

Comment:
```suggestion
      https://api.github.com/repos/"${2}"/"${3}"/"${4}" > "${6}"
```
Double quote to prevent globbing and word splitting.